### PR TITLE
Merge Develop Branch to Master

### PR DESCRIPTION
**Pull request by ajimae**

### CODE REVIEW

File: packages/history-sdk/package.json [hunk 1]
```diff
@@ -23,11 +23,11 @@
     "./dist/commercetools-history-sdk.esm.js": "./dist/commercetools-history-sdk.browser.esm.js"
   },
   "dependencies": {
-    "@commercetools/sdk-client": "^2.1.1",
+    "@commercetools/sdk-client": "^2.1.2",
     "@commercetools/sdk-middleware-auth": "^6.0.4",
     "@commercetools/sdk-middleware-http": "^6.0.4",
     "@commercetools/sdk-middleware-logger": "^2.1.1",
```
Comment (jenschude) [inline packages/history-sdk/package.json:29]: as i just go through the cahnges

shouldn't this be changed to the sdk-client-v2 ?

Also the different middlewares are already part of the sdk-client-v2 so not needed anymore here. Same applies to the other sdk packages

**Review by jenschude — CHANGES_REQUESTED** [review]

File: packages/sdk-client/package.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/commercetools/commercetools-sdk-typescript.git"
+  },
+  "bugs": {
+    "url": "https://github.com/commercetools/commercetools-sdk-typescript/issues"
+  },
+  "dependencies": {
+    "@commercetools/platform-sdk": "^1.17.1",
```
Comment (jenschude) [inline packages/sdk-client/package.json:32]: This should not be a dependency as platform SDK is dependent on the client package. Else you run in a circular dependency

Comment (ajimae) [reply packages/sdk-client/package.json:32]: We need this for a CI build before we can run tests, if I should remove it, the build will fail exit, hence test won't even run.

File: packages/history-sdk/package.json [hunk 3]
```diff
@@ -23,14 +23,11 @@
     "./dist/commercetools-history-sdk.esm.js": "./dist/commercetools-history-sdk.browser.esm.js"
   },
   "dependencies": {
-    "@commercetools/sdk-client": "^2.1.1",
-    "@commercetools/sdk-middleware-auth": "^6.0.4",
-    "@commercetools/sdk-middleware-http": "^6.0.4",
-    "@commercetools/sdk-middleware-logger": "^2.1.1",
-    "querystring": "0.2.1"
+    "querystring": "^0.2.1"
```
Comment (jenschude) [inline packages/history-sdk/package.json:26]:
```suggestion
    "@commercetools/sdk-client-v2": "^0.0.1",
    "querystring": "^0.2.1"
```

File: packages/history-sdk/package.json [hunk 4]
```diff
@@ -23,14 +23,11 @@
...
   },
   "devDependencies": {
     "@commercetools/platform-sdk": "1.17.1",
+    "@commercetools/sdk-client-v2": "^0.0.1",
```
Comment (jenschude) [inline packages/history-sdk/package.json:30]:
```suggestion
```

File: packages/importapi-sdk/package.json [hunk 5]
```diff
@@ -23,13 +23,10 @@
     "./dist/commercetools-importapi-sdk.esm.js": "./dist/commercetools-importapi-sdk.browser.esm.js"
   },
   "dependencies": {
-    "@commercetools/sdk-client": "^2.1.1",
-    "@commercetools/sdk-middleware-auth": "^6.0.4",
-    "@commercetools/sdk-middleware-http": "^6.0.4",
-    "@commercetools/sdk-middleware-logger": "^2.1.1",
-    "querystring": "0.2.1"
+    "querystring": "^0.2.1"
```
Comment (jenschude) [inline packages/importapi-sdk/package.json:26]:
```suggestion
    "@commercetools/sdk-client-v2": "^0.0.1",
    "querystring": "^0.2.1"
```

File: packages/importapi-sdk/package.json [hunk 6]
```diff
@@ -23,13 +23,10 @@
...
   },
   "devDependencies": {
+    "@commercetools/sdk-client-v2": "^0.0.1",
```
Comment (jenschude) [inline packages/importapi-sdk/package.json:29]:
```suggestion
```

File: packages/ml-sdk/package.json [hunk 7]
```diff
@@ -23,14 +23,11 @@
     "./dist/commercetools-ml-sdk.esm.js": "./dist/commercetools-ml-sdk.browser.esm.js"
   },
   "dependencies": {
-    "@commercetools/sdk-client": "^2.1.1",
-    "@commercetools/sdk-middleware-auth": "^6.0.4",
-    "@commercetools/sdk-middleware-http": "^6.0.4",
-    "@commercetools/sdk-middleware-logger": "^2.1.1",
-    "querystring": "0.2.1"
+    "querystring": "^0.2.1"
```
Comment (jenschude) [inline packages/ml-sdk/package.json:26]:
```suggestion
    "@commercetools/sdk-client-v2": "^0.0.1",
    "querystring": "^0.2.1"
```

File: packages/ml-sdk/package.json [hunk 8]
```diff
@@ -23,14 +23,11 @@
...
   },
   "devDependencies": {
     "@commercetools/platform-sdk": "1.17.1",
+    "@commercetools/sdk-client-v2": "^0.0.1",
```
Comment (jenschude) [inline packages/ml-sdk/package.json:30]:
```suggestion
```

File: packages/platform-sdk/package.json [hunk 9]
```diff
@@ -23,13 +23,10 @@
     "./dist/commercetools-platform-sdk.esm.js": "./dist/commercetools-platform-sdk.browser.esm.js"
   },
   "dependencies": {
-    "@commercetools/sdk-client": "^2.1.1",
-    "@commercetools/sdk-middleware-auth": "^6.0.4",
-    "@commercetools/sdk-middleware-http": "^6.0.4",
-    "@commercetools/sdk-middleware-logger": "^2.1.1",
-    "querystring": "0.2.1"
+    "querystring": "^0.2.1"
```
Comment (jenschude) [inline packages/platform-sdk/package.json:26]:
```suggestion
    "@commercetools/sdk-client-v2": "^0.0.1",
    "querystring": "^0.2.1"
```

File: packages/platform-sdk/package.json [hunk 10]
```diff
@@ -23,13 +23,10 @@
...
   },
   "devDependencies": {
+    "@commercetools/sdk-client-v2": "^0.0.1",
```
Comment (jenschude) [inline packages/platform-sdk/package.json:29]:
```suggestion
```

**Review by jenschude — APPROVED** [review]
The sdk client should be a direct dependency for all sdk packages

**Pull request merged**